tests/test_cli.py

```diff
@@ -5,20 +5,55 @@
 from .common import get_test_data
 
 
-def test_generate_grid_weights():
+def test_generate_grid_weights_with_nc_input_and_2d_coords():
     runner = CliRunner()
-    ds = map(
-        str,
-        [
-            get_test_data("raven-routing-sample", "finalcat_hru_info.zip")[0],
-            get_test_data("raven-routing-sample", "VIC_streaminputs.nc")[0],
-        ],
-    )
+    params = [
+        get_test_data("raven-routing-sample", "VIC_streaminputs.nc")[0],
+        get_test_data("raven-routing-sample", "finalcat_hru_info.zip")[0],
+    ]
+    params = map(str, params)
 
-    result = runner.invoke(generate_grid_weights, ds)
+    result = runner.invoke(generate_grid_weights, params)
 
     assert result.exit_code == 0
     assert not result.exception
     assert ":NumberHRUs 51" in result.output
     assert ":NumberGridCells 100" in result.output
     assert len(result.output.split("\n")) == 216
+
+
```

Comment:
I think we need an additional test where we invoke multiple gauges to check that removing duplicates etc is working properly.